### DISCUSSION TITLE
[question] Migrating from zwavej2 to zwave2mqtt

### DISCUSSION THREAD

**Discussion by christianreiss** [zwave-js/zwave-js-ui#2356] [General]
Hey folks,

I am running a rather large installation of Homeassistant here; along with 120-or-so zwave nodes. Everything is running on a real server (no raspberry pi) inside docker containers. Currently the official zwavejs docker container is running. 

Now I have no issues with this at all. It is working like a charm. The only itch I have is to backup (and restore) the NVR part of the stick. I really, really, really don't want to go though all the rooms in all the stories of my house and open up all the outlets, switches, cover shutters to re-bind them all. I want/need a binding backup. 

I know that zwave2mqtt has such an option to backup/restore NVR. I know because I used it... Getting ahead of myself. 
So I did my homework as good as I could, and found out the current way to switch from zwavejs to zwave2mqtt is (or looks like):

1. Take a VM Snapshot.
2. Remove zwave Integration from Homeassistant,
3. Remove the old zwavejs docker container,
4. copy the cache data over to a save place,
5. note the encryption keys,
6. Launch zwave2mqtt docker container,
7. let it run for some,
8. shut it down again,
9. Copy the cache files over (Unsure: Is the name the same?)
10. Start the container again with exposed port 3000,
11. re-install the zwave integration.

I have two issues with this:
- All nodes have different names,
- I am unsure if the cache files have the same names.

Also, nodes.json does not exist in zwavejs volumes.

Or maybe I am doing this wrong alltogether. What's a better way to move from zwave to zwave2mqtt while keeping the nodes in homeassistant?

PS: I was able to see all nodes in the webinterface. And the integration had all the entities, too. Just not correctly named. I feel I am just a bit off to get this right.

This is also where I snatched a NVR backup before rolling the snapshot back.

ANY help is greatly appreciated!
-Chris.

**Comment by robertsLando** [comment]
@christianreiss How did you installed zwavejs2mqtt? As an hass addon or as a separete docker container? How did you setup the store volume? Zwavejs2mqtt stores everything in `store` folder that needs to be mapped in a dedicated volume (local folder or docker volume). nodes.json will be there. About your naming issue, I dunno how hass handels them so I tag someone that could help you with them:


@kpine 
@raman325 

On my side the only thing you can do to set node names and location on startup is to set them from the ui or edit the nodes.json file you can find in store folder to match the names/locations you are using now

**Comment by christianreiss** [comment]
Hey,

Thanks for replying. I did install it via the community add-on (resulting in a docker container). There is an automatic volume-mapping done, and was able to see the directory structure for zwave2mqtt. 

Come to think of it, I think I might have copied the files into the wrong dir:

```
/root/1/usr/share/hassio/addons/data/a0d7b954_zwavejs2mqtt/cache/c95f13ce.json
/root/1/usr/share/hassio/addons/data/a0d7b954_zwavejs2mqtt/store/c95f13ce.json
```
(Ignore the beginning of the path, thats a cloned vm running a rescuecd).

So I place everything into store, not cache? Do the filenames keep the same?

**Comment by robertsLando** [comment]
I dunno what is `cache` but the correct one is  `store`. Also I'm not sure that cache file is something from zwavejs or else? What is the content of it?

**Comment by christianreiss** [comment]
This is the zwavejs volume dir:

```
[root@sysrescue ~/1/usr/share/hassio/addons/data/a0d7b954_zwavejs2mqtt]# ls -lha /root/1/root/core_zwave_js/cache/
total 1.8M
drwxr-xr-x 4 root root 4.0K Apr  6 06:27 .
drwxr-xr-x 4 root root 4.0K Mar 23 09:35 ..
-rw-r--r-- 1 root root 363K Mar 23 09:34 c95f13ce.json
-rw-r--r-- 1 root root 773K Apr  6 06:36 c95f13ce.metadata.jsonl
drwxr-xr-x 2 root root 4.0K Apr  6 06:39 c95f13ce.metadata.jsonl.lock
-rw-r--r-- 1 root root 670K Apr  6 06:37 c95f13ce.values.jsonl
drwxr-xr-x 2 root root 4.0K Apr  6 06:39 c95f13ce.values.jsonl.lock
```

And cache/store of zwave2mqtt:

```[root@sysrescue ~/1/usr/share/hassio/addons/data/a0d7b954_zwavejs2mqtt]# ls -lha cache/
total 1.8M
drwxr-xr-x 4 root root 4.0K Apr  6 06:27 .
drwxr-xr-x 5 root root 4.0K Apr  6 07:08 ..
-rw-r--r-- 1 root root 363K Mar 23 09:34 c95f13ce.json
-rw-r--r-- 1 root root 773K Apr  6 06:36 c95f13ce.metadata.jsonl
drwxr-xr-x 2 root root 4.0K Apr  6 06:39 c95f13ce.metadata.jsonl.lock
-rw-r--r-- 1 root root 670K Apr  6 06:37 c95f13ce.values.jsonl
drwxr-xr-x 2 root root 4.0K Apr  6 06:39 c95f13ce.values.jsonl.lock
[root@sysrescue ~/1/usr/share/hassio/addons/data/a0d7b954_zwavejs2mqtt]# ls -lha store/
total 2.3M
drwxr-xr-x 5 root root 4.0K Apr  6 07:10 .
drwxr-xr-x 5 root root 4.0K Apr  6 07:08 ..
-rw-r--r-- 1 root root 363K Apr  6 07:08 c95f13ce.json
-rw-r--r-- 1 root root 759K Apr  6 07:10 c95f13ce.metadata.jsonl
drwxr-xr-x 2 root root 4.0K Apr  6 07:10 c95f13ce.metadata.jsonl.lock
-rw-r--r-- 1 root root 669K Apr  6 07:10 c95f13ce.values.jsonl
drwxr-xr-x 2 root root 4.0K Apr  6 07:10 c95f13ce.values.jsonl.lock
-rw-r--r-- 1 root root    3 Apr  6 06:41 nodes.json
-rw-r--r-- 1 root root 512K Apr  6 07:06 NVM_2022-04-06.bin
drwxr-xr-x 2 root root 4.0K Apr  6 06:41 sessions
-rw-r--r-- 1 root root  980 Apr  6 06:49 settings.json
-rw-r--r-- 1 root root  186 Apr  6 06:41 users.json
```

So the c95f13ce* files originated my rsync'ing the files in there. They _seem_ to be accessed and modified, so right names then?

```[root@sysrescue ~/1/usr/share/hassio/addons/data/a0d7b954_zwavejs2mqtt]# md5sum cache/c95f13ce.metadata.jsonl store/c95f13ce.metadata.jsonl /root/1/root/core_zwave_js/cache/c95f13ce.metadata.jsonl
a80576230e469af20d1c5850c708d93f  cache/c95f13ce.metadata.jsonl
17b6676219e4d6e579f1ac57488eaa6e  store/c95f13ce.metadata.jsonl
a80576230e469af20d1c5850c708d93f  /root/1/root/core_zwave_js/cache/c95f13ce.metadata.jsonl
```

Question remain: Why did everything not plugin-out-plug-in work in Homeassistant? New names?

**Comment by robertsLando** [comment]
> So the c95f13ce* files originated my rsync'ing the files in there. They seem to be accessed and modified, so right names then?

Yep, now they are in the correct folder.

**Comment by kpine** [comment]
> 2. Remove zwave Integration from Homeassistant,

(I'm assuming you mean zwave_js integration.)

If you remove the integration you lose all of your naming and customization. **Don't do that.** Just reinstall it at the end to change the URL.

**Reply by christianreiss** [reply]
True, true. That might solve my issues. I did see all the nodes in the gui earlier on.

But keeping the integration online (zwave_js/ docker/ ha) will keep spinning up the zwave_js docker container every few seconds. Only by ... oh could a reinstall also solve the always-spinning up the zwave_js docker container issue, too?

**Reply by kpine** [reply]
I don't believe keeping the integration online will restart or re-install the Z-Wave JS addon. AFAIK, that is only checked when the integration is started. If you are worried about that, just disable the integration, as the [**official instructions**](https://www.home-assistant.io/integrations/zwave_js/#how-do-i-switch-between-the-official-z-wave-js-add-on-and-zwavejs2mqtt) describe. Of course, you must correctly re-configure the integration as the instructions tell you to.

**Comment by christianreiss** [comment]
With all the help this was solved trivialy. Steps taken:

1. Snapshot!
3. Disable zwave integration
4. Ensure old zwave addon is stopped
5. Install zwave2mqtt addon
6. configure zwave2mqtt addon (security keys, device), export port 3000
7. Stop zwave2mqtt integration
8. Copy files over, into store-dir (from zwavejs to zwave2mqtt)
9. Add the Z-Wave JS integration again (even though it is still installed), and uncheck the “Use the Z-Wave JS Supervisor add-on”. 
10. Uninstall the official Z-Wave JS add-on.
11. Enable the Z-Wave JS integration.
12. Profit!
